lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -120,8 +120,20 @@ func (m *PortlayerEventMonitor) Start() error {
 	}
 
 	m.stop = make(chan struct{})
-	go m.monitor()
-
+	go func() {
+		var err error
+		for {
+			select {
+			case <-m.stop:
+				break
+			default:
+				if err = m.monitor(); err != nil {
+					log.Error(err)
+				}
+				log.Warn("Restarting Portlayer event monitor")
```

Comment:
Will be logged even if we're not going to restart. Perhaps we should just combine this with the error logging.

Reply:
oh right, didn't think about it after we restructured the loop.